### PR TITLE
[Python] List sort with 

### LINKED ISSUE #3638: [Python] Array sort functions not implemented
### Description

Array sorting functions fail.

#### test.fsx
```fsharp
Array.sort [|3;2;4|]
Array.sortDescending [|3;2;4|]
```
#### test.cmd
```shell
dotnet fable . --lang python
python test.py
```

### Expected and actual results

Expected result: should return sorted arrays

Actual result: `AttributeError: 'array.array' object has no attribute 'sort'`

### Related information

* `dotnet fable --version`: 4.5.0
* `dotnet tool list/update/install`: 4.6.0
* Windows11

@freymaurer

### LINKED ISSUE #3649: [Python] List.sortBy just sorts by the unmapped values
### Description

List.sortBy basically returns the same results as List.sort

#### test.fsx
```fsharp
[2;3;7;1;4]
|> List.sortBy (fun x -> x % 2)
|> printfn "%O"
```
#### test.cmd
```shell
dotnet fable . --lang python
python test.py
```

### Expected and actual results

Expected result: `[2; 4; 3; 7; 1]`

Actual result: `[1; 2; 3; 4; 7]`

### Related information

* `dotnet fable --version`: 4.5.0
* `dotnet tool list/update/install`: 4.7.0
* Windows11

@freymaurer

### PR DESCRIPTION
**Pull request by dbrattli**
Make own copies of `List.fs` and `Array.fs` so we can fix the current we have problems with sorting for Python. Not sure if the intention was to have a single implementation for all languages, but see that Rust already have separate implementations.

Also imports `Callable` from `collections.abc` which is the preferred way instead of `typing`.

Fixes #3649
Fixes #3638

### CODE REVIEW

File: src/Fable.Cli/CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 #### Python
 
 * [GH-3645](https://github.com/fable-compiler/Fable/pull/3645) Add `TimeSpan.Parse` and `TimeSpan.TryParse` support to Python (by @MangelMaxime)
+  [GH-3649](https://github.com/fable-compiler/Fable/issues/3649) Fixes for `List.sortBy` (by @dbrattli)
+  [GH-3638](https://github.com/fable-compiler/Fable/issues/3638) Fixes for `Array.sort` and `Array.sortDescending` (by @dbrattli)
```
Comment (MangelMaxime) [inline src/Fable.Cli/CHANGELOG.md:21]: This is missing some `*` at the beginning of the line

File: src/Fable.Cli/CHANGELOG.md [hunk 2]
```diff
@@ -25,8 +27,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 #### All
 
 * Overall performance improvements
-    * [GH-3620](https://github.com/fable-compiler/Fable/pull/3620) Removed double-dictionary lookups (by @Thorium)
-    * [GH-3624](https://github.com/fable-compiler/Fable/pull/3624) Add G-Research analyzers and fix reported issues (by @nojaf)
+  * [GH-3620](https://github.com/fable-compiler/Fable/pull/3620) Removed double-dictionary lookups (by @Thorium)
+  * [GH-3624](https://github.com/fable-compiler/Fable/pull/3624) Add G-Research analyzers and fix reported issues (by @nojaf)
```
Comment (MangelMaxime) [inline src/Fable.Cli/CHANGELOG.md:31]: Can we please keep the 4 spaces indentation?

There are some other elements below using 4 spaces.

**Review by MangelMaxime — CHANGES_REQUESTED** [review]

File: src/Fable.Cli/CHANGELOG.md (hunk 2 above)
Comment (dbrattli) [reply src/Fable.Cli/CHANGELOG.md]: Ok, perhaps we can define some markdownlint rules for the file. Currently there's multiple issues e.g: 
Screenshot 2023-12-12 at 09 37 57

Comment (MangelMaxime) [reply src/Fable.Cli/CHANGELOG.md]: We could but nothing guarantee that everyone is going to use VSCode and see the linter information.

It seems like the extension you are using also has a CLI, so to enforce it we could make a pre-commit hooks to enforce CHANGELOG.md formatting.

Is this a good direction to go with? For Fable repositiory,

I came up with the following configuration:

```json
{
    "MD024": false, // Allow duplicates headings
    "MD007": {
        "indent": 4 // Use 4 spaces indentations
    },
    "MD038": false, // Allow space in code snippet, no idea why this is on by default
    "MD013": false // Disable line-length checks we are now using URLs in the changelog and that increase the line length considerably (but make links works on external websites like NuGet, etc)
}
```

**Pull request merged**